seqr/views/utils/dataset_utils.py

```diff
@@ -298,67 +320,103 @@ def _parse_rna_row(row, column_map, required_column_map, missing_required_fields
         for mapped_key, format_func in (format_fields or {}).items():
             row_dict[mapped_key] = format_func(row_dict[mapped_key])
 
-        missing_cols = [col_id for col, col_id in required_column_map.items() if not row.get(col)]
+        missing_cols = {col_id for col, col_id in required_column_map.items() if not row.get(col)}
+        if allow_missing_gene:
+            missing_cols.discard(GENE_ID_COL)
         sample_id = row_dict.pop(SAMPLE_ID_COL) if SAMPLE_ID_COL in row_dict else row[SAMPLE_ID_COL]
         if missing_cols:
             for col in missing_cols:
                 missing_required_fields[col].append(sample_id)
-        if not missing_cols or (allow_missing_gene and missing_cols == [GENE_ID_COL]):
+        if not missing_cols:
             yield sample_id, row_dict
 
 
-def _load_rna_seq_file(file_path, user, column_map, mapping_file=None, get_unique_key=None, allow_missing_gene=False, **kwargs):
+def _load_rna_seq_file(
+        file_path, user, potential_loaded_samples, update_sample_models, save_sample_data, get_matched_sample, mismatches,
+        column_map, mapping_file=None, get_unique_key=None, allow_missing_gene=False, ignore_extra_samples=False, **kwargs,
+):
 
     sample_id_to_individual_id_mapping = {}
     if mapping_file:
         sample_id_to_individual_id_mapping = load_mapping_file_content(mapping_file)
 
-    samples_by_id = defaultdict(dict)
+    samples_by_guid = defaultdict(dict)
     f = file_iter(file_path, user=user)
-    header = _parse_tsv_row(next(f))
+    parsed_f = parse_file(file_path.replace('.gz', ''), f, iter_file=True)
```

Comment:
ok I'm reviewing generators and iterators and yield in python as we speak so this might be a silly question, but why do we need to `yield` the parsed line (`file_utils.py` line 38) if the un-parsed line is already `yield`ed from `file_iter`?

Reply:
I'm not seeing the unparsed line being yielded, just the parsed line

Reply:
here's the unparsed yield line I'm referring to (in `file_iter()`) https://github.com/broadinstitute/seqr/blob/b57c19e2615fd7669e2f27d046ff5c2e4b6f667a/seqr/utils/file_utils.py#L68

Reply:
Ah. So that part is a generator/iterator, but then what happens is that generator is passed into `parse_file` and then the whole thing is read into a list, so the whole thing is then in memory and no longer an iterator:
https://github.com/broadinstitute/seqr/blob/b57c19e2615fd7669e2f27d046ff5c2e4b6f667a/seqr/views/utils/file_utils.py#L46

So what this change is doing is saying that in this case we do not want to actually evaluate the whole iterator here, we want to continue to keep it as a generator/ iterator (I sometimes am fuzzy on the distinction between those)

Reply:
ahhhhh I see thanks for the clarification